generators/client/templates/angular/src/main/webapp/app/layouts/profiles/_profile.service.ts

```diff
@@ -28,19 +28,19 @@ export class ProfileService {
     private profileInfoUrl = SERVER_API_URL + 'api/profile-info';
     private profileInfo: Promise<ProfileInfo>;
 
-    constructor(private http: Http) { }
+    constructor(private http: HttpClient) { }
 
     getProfileInfo(): Promise<ProfileInfo> {
         if (!this.profileInfo) {
-            this.profileInfo = this.http.get(this.profileInfoUrl)
-                .map((res: Response) => {
-                    const data = res.json();
-                    const pi = new ProfileInfo();
-                    pi.activeProfiles = data.activeProfiles;
-                    pi.ribbonEnv = data.ribbonEnv;
-                    pi.inProduction = data.activeProfiles.includes('prod');
-                    pi.swaggerEnabled = data.activeProfiles.includes('swagger');
-                    return pi;
+            this.profileInfo = this.http.get<ProfileInfo>(this.profileInfoUrl, { observe: 'response' })
+            .map((res: HttpResponse<ProfileInfo>) => {
+                const data = res.body;
+                const pi = new ProfileInfo();
+                pi.activeProfiles = data.activeProfiles;
+                pi.ribbonEnv = data.ribbonEnv;
+                pi.inProduction = data.activeProfiles.indexOf('prod') !== -1;
```

Comment:
why switched from `includes`? please use includes as its easier to comprehend than indexOf

Reply:
it seems like its not working with typed instances with before ES7 https://github.com/gdi2290/angular-starter/issues/931 
as includes is not implemented for string[] < ES7

Reply:
https://github.com/Microsoft/TypeScript/issues/2340#issuecomment-316664621   this states it as well i could change to es7 but don't know what side effects this would have

Reply:
Dont worry about it, it only breaks in IE11 and for that we recently merged a PR adding polyfill for es7 array

Reply:
maybe i am just stupid but my build always fails with: Property 'includes' does not exist on type 'string[]'

Reply:
I fixed it, since its an es7 feature we need to set lib version as es7 in tsconfig